client/state/data-layer/wpcom/checklist/index.js

```diff
@@ -29,8 +29,26 @@ export const fetchChecklist = action =>
 		action
 	);
 
-export const receiveChecklistSuccess = ( action, checklist ) =>
-	receiveSiteChecklist( action.siteId, checklist );
+export const receiveChecklistSuccess = ( action, receivedChecklist ) => {
```

Comment:
When I [suggested this should be in `fromApi`](https://github.com/Automattic/wp-calypso/pull/34050#discussion_r294706429) I mean the `dispatchRequest fromApi` which I've just found is rather poorly documented.

```diff
diff --git a/client/state/data-layer/wpcom/checklist/index.js b/client/state/data-layer/wpcom/checklist/index.js
index 8c68983ae4..4335543a8d 100644
--- a/client/state/data-layer/wpcom/checklist/index.js
+++ b/client/state/data-layer/wpcom/checklist/index.js
@@ -29,15 +29,12 @@ export const fetchChecklist = action =>
 		action
 	);
 
-export const receiveChecklistSuccess = ( action, receivedChecklist ) => {
-	let checklist = receivedChecklist;
-
-	// Legacy object-based data format, let's convert it to the new array-based format and ultimately remove it.
-	if ( ! Array.isArray( receivedChecklist.tasks ) ) {
-		checklist = {
-			...receivedChecklist,
-			tasks: Object.keys( receivedChecklist.tasks ).map( taskId => {
-				const { completed, ...rest } = receivedChecklist.tasks[ taskId ];
+const fromApi = data => {
+	if ( ! Array.isArray( data.tasks ) ) {
+		return {
+			...data,
+			tasks: Object.keys( data.tasks ).map( taskId => {
+				const { completed, ...rest } = data.tasks[ taskId ];
 				return {
 					id: taskId,
 					isCompleted: completed,
@@ -47,13 +44,17 @@ export const receiveChecklistSuccess = ( action, receivedChecklist ) => {
 		};
 	}
 
-	return receiveSiteChecklist( action.siteId, checklist );
+	return data;
 };
 
+export const receiveChecklistSuccess = ( action, checklist ) =>
+	receiveSiteChecklist( action.siteId, checklist );
+
 const dispatchChecklistRequest = dispatchRequest( {
 	fetch: fetchChecklist,
 	onSuccess: receiveChecklistSuccess,
 	onError: noop,
+	fromApi,
 } );
 
 export const updateChecklistTask = action =>
@@ -74,6 +75,7 @@ const dispatchChecklistTaskUpdate = dispatchRequest( {
 	fetch: updateChecklistTask,
 	onSuccess: receiveChecklistSuccess,
 	onError: noop,
+	fromApi,
 } );
 
 registerHandlers( 'state/data-layer/wpcom/checklist/index.js', {
```